src/common/ceph_crypto.cc

```diff
@@ -83,6 +84,7 @@ void ceph::crypto::shutdown()
   assert(crypto_refs > 0);
   if (--crypto_refs == 0) {
     NSS_ShutdownContext(crypto_context);
+    PR_Cleanup();
```

Comment:
@liewegas we always have a `g_ceph_context`, and `common_init_finish()` calls `cct->init_crypto()` and we have a smart pointer managing the life cycle of the global `g_ceph_context` (remember the huge patch introducing the smart pointer in basically all `main()`s?), so the last one who releases `crypto_context` is always the destructor of that smart pointer in `main()`. so it's always the primordial thread.